debian/entrypoint.sh

```diff
@@ -64,9 +56,9 @@ if [ -n "${api_key}" -o -n "${controller_hostname}" -o -n "${controller_url}" -o
     sh -c "sed -i.old -e 's/api_key.*$/api_key = $api_key/' \
 	${agent_conf_file}"
 
-    test -n "${controller_hostname}" && \
-    echo " ---> using hostname = ${controller_hostname}" && \
-    sh -c "sed -i.old -e 's/instance_name.*$/instance_name = $controller_hostname/' \
+    test -n "${instance_name}" && \
```

Comment:
wouldn't this always be true if we set `instance_name=$(hostname -f)` ?

Reply:
entrypoint.sh should be taking instance_name (if provided with the run command) and writing that to agent.conf before launching agent.
entrypoint.sh should not be setting instance_name to hostname.  Let agent behavior handle that behavior - so we don’t have two different code bases trying to mimic each other.

Reply:
ok, so the current master logic is the following:
1. If the user specifies `--hostname` param when running docker, it will be returned by the `hostname -f` command
2. the `hostname -f` command is used as our default value for variable `controller_hostname`
3. value of `controller_hostname` can also be overridden using an `ENV_CONTROLLER_HOSTNAME` environment
4. the value of `controller_hostname` will be set in agent.conf to be the value of `instance_name` (so at this point the `ENV_CONTROLLER_HOSTNAME` is redundant, because we are accomplishing the same using `--hostname`, which is also described in the README as the only way to set `instance_name`)

how I see the changes after your comment:
1. we read the value of `INSTANCE_NAME` (or `ENV_INSTANCE_NAME` if we want to be consistent with our naming) inside entrypoint
2. if it's not empty, we set it to be the value of `instance_name` inside agent.conf
3. If it's empty, we do nothing - there is no defaulting to `hostname -f`
4. we do not make use of `hostname -f` in any way inside the entrypoint
5. `ENV_CONTROLLER_HOSTNAME` is removed

@brianehlert Is my understanding correct?